### PR TITLE
poetry2nix: 1.41.0 -> 1.42.0

### PR DESCRIPTION
**Pull request by adisbladis**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/poetry2nix/poetry2nix/default.nix [hunk 1]
```diff
@@ -17,7 +17,7 @@ let
   toPluginAble = (import ./plugins.nix { inherit pkgs lib; }).toPluginAble;
 
   # List of known build systems that are passed through from nixpkgs unmodified
-  knownBuildSystems = lib.importJSON ./known-build-systems.json;
+  knownBuildSystems = builtins.fromJSON (builtins.readFile ./known-build-systems.json);
```
Comment (SuperSandro2000) [inline pkgs/development/tools/poetry2nix/poetry2nix/default.nix:20]: Can you please upstream this? Thanks

File: pkgs/development/tools/poetry2nix/poetry2nix/overrides/default.nix [hunk 2]
```diff
@@ -212,7 +212,7 @@ lib.composeManyExtensions [
             nativeBuildInputs = with pkgs;
               (old.nativeBuildInputs or [ ])
                 ++ lib.optionals (lib.versionAtLeast old.version "4")
-                [ rustc cargo rustPlatform.cargoSetupHook self.setuptools-rust ];
+                (with pkgs.rustPlatform; [ rust.rustc rust.cargo cargoSetupHook self.setuptools-rust ]);
```
Comment (SuperSandro2000) [inline pkgs/development/tools/poetry2nix/poetry2nix/overrides/default.nix:215]: same here with the rust changes

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
drafting, feel free to undraft when eval is fixed

**Comment by Izorkin** [comment]
After this update, the nixops 1.7 package is no building. How can I fix it?

**Pull request merged**